### PR TITLE
feat(si): Implement a Container Engine Trait

### PR DESCRIPTION
**Pull request by stack72**
This is a container engine trait that needs to be completed before any new containe engines are added to the SI Launcher

prerequisite for #2607

### CODE REVIEW

**Comment by vwbusguy** [comment]
This conditional block needs removed for it to work: https://github.com/systeminit/si/blob/dad5771e4321d5eea250cf750a6752b2541aaa16/bin/si/src/main.rs#L86

File: bin/si/src/main.rs [hunk 1]
```diff
@@ -23,10 +25,9 @@ async fn main() -> Result<()> {
...
     let is_preview = args.is_preview;
 
-    let docker_sock = if let Some(sock) = args.docker_sock.clone() {
-        sock
-    } else {
-        "".to_string()
+    let engine = match args.engine() {
+        Engine::Docker => DockerEngine::new(args.docker_sock.clone()).await?,
+        Engine::Podman => PodmanEngine::new(args.docker_sock.clone()).await?,
     };
```
Comment (nickgerace) [inline bin/si/src/main.rs:31]: I love this!

File: lib/si-cli/src/engine.rs
Comment (nickgerace) [inline lib/si-cli/src/engine.rs:1]: Way to go!!

**Review by nickgerace — APPROVED** [review]
Congrats! Good idea for a trait.

**Comment by stack72** [comment]
bors merge

**Pull request merged**